woocommerce/woocommerce-gateway-stripe
# Create backend classes for express checkout element

This PR prepares the backend for #3404 and #3403 

## Changes proposed in this Pull Request:
- Created the PHP classes for the express checkout element.
    - `WC_Stripe_Express_Checkout_Element`
    - `WC_Stripe_Express_Checkout_Ajax_Handler`
    - `WC_Stripe_Express_Checkout_Helper`
- All the functions in the above classes are copied from `WC_Stripe_Payment_Request` [class](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php), just broken down to three separate classes.
    - Discarded a couple of unnecessary checks ([example](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L1781)).
    - Discarded the [branded button](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L295) and [custom button](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L317) related functions which we no longer offer in the settings page.

## Testing instructions
- Run `npm install`, `npm start`.
- Enable Google/Apple Pay from the Stripe settings page.
- As a shopper, add a product to your cart and go to the checkout page. Confirm that Google/Apple Pay button loads on both shortcode and block checkout the pages.

<img width="693" alt="Screenshot 2024-09-11 at 2 32 05 PM" src="https://github.com/user-attachments/assets/e88c64ec-c356-4115-b9fd-513f5709b790">

- Check the JS scripts loaded in the network tab and confirm that `assets/js/stripe-payment-request.js` is loaded and `build/express_checkout.js` is not loaded.
- Now enable the ECE feature flag (return true from `is_stripe_ece_enabled` function or in your database add an option `_wcstripe_feature_ece` and set `yes` as the value).
- Refresh the checkout page. Confirm that `build/express_checkout.js` script is loaded but `assets/js/stripe-payment-request.js` is not loaded.
- Confirm that Google/Apple Pay button loads on the block checkout page only.

<img width="693" alt="Screenshot 2024-09-11 at 2 30 00 PM" src="https://github.com/user-attachments/assets/42c5b1fe-f841-40e2-8324-d62e26ef67dd">
